plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcPageSink.java

```diff
@@ -46,10 +47,14 @@
 
     private final List<Type> columnTypes;
     private final List<WriteFunction> columnWriters;
+    private final int maxBatchSize;
     private int batchSize;
 
-    public JdbcPageSink(ConnectorSession session, JdbcOutputTableHandle handle, JdbcClient jdbcClient)
+    public JdbcPageSink(ConnectorSession session, JdbcOutputTableHandle handle, JdbcClient jdbcClient, int maxBatchSize)
     {
+        checkArgument(maxBatchSize > 0, "max batch size must be positive: %s", maxBatchSize);
```

Comment:
Should we have a reasonable upper-limit too to make sure memory usage on the worker has a ceiling? Just a thought - no idea if it matters.

WDYT @wendigo @kokosing ?

Reply:
But what would be the upper-limit, it could be based on size of n Pages instead of number of rows. Just a thought

Reply:
Yeah. It should be added to the config class and session property provider validation so it's get validated there as well

Reply:
done